cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/PartitionConsumer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib.partitioned;
+
+import co.cask.cdap.api.dataset.lib.Partition;
+import co.cask.cdap.api.dataset.lib.PartitionDetail;
+import co.cask.cdap.api.dataset.lib.PartitionedFileSet;
+
+import java.util.List;
+
+/**
+ * Incrementally consumes new/unprocessed {@link Partition}s of a {@link PartitionedFileSet}.
+ */
+public interface PartitionConsumer {
+
+  /**
+   * @return list of {@link PartitionDetail}s that have not yet processed.
+   */
+  List<PartitionDetail> consumePartitions();
+
+  /**
+   * @param limit upper limit on number of partitions to consume
+   * @return list of {@link PartitionDetail}s that have not yet processed.
+   */
+  List<PartitionDetail> consumePartitions(int limit);
+
+  /**
+   * This method must be called on any partitions returned by the {@code #consumePartitions} method.
+   *
+   * @param partitions list of partitions to mark as either succeeded or failed processing
+   * @param succeeded whether or not processing of the specified partitions was successful
+   */
+  void onFinish(List<? extends Partition> partitions, boolean succeeded);
```

Comment:
added this to the javadoc:
* If a program fails to call this method for any partitions, those partitions will be 'expired' after a timeout
   * defined on the configured {@link ConsumerConfiguration}.